### PR TITLE
Splits Kit

### PR DESCRIPTION
**Pull request by shrimpynuts**
https://www.npmjs.com/package/@0xsplits/splits-kit/

Apologies this is one huge PR, but wasn't sure of better method in this case.
Mostly want eyes on config files (esp package.json, tsconfig), if anything looks off.

### CODE REVIEW

File: packages/splits-sdk-react/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@0xsplits/splits-sdk-react",
-  "version": "1.0.6",
```
Comment (shrimpynuts) [inline packages/splits-sdk-react/package.json:3]: Sorry I accidentally committed a `lerna-version` call along the way

File: packages/splits-kit/tsconfig.json [hunk 2]
```diff
@@ -2,11 +2,9 @@
   "extends": "../../tsconfig.json",
   "baseUrl": "./src",
   "compilerOptions": {
+    "module": "ESNext",
```
Comment (shrimpynuts) [inline packages/splits-kit/tsconfig.json:5]: I was running into an error importing this library, which was a result of using wagmi (an es module) in a commonjs `require("wagmi")` statement. 

Making our library compile to es module fixed this issue.

I believe this is the way to go for javascript run in the browser.

For more info:
https://www.tsmean.com/articles/learn-typescript/typescript-module-compiler-option/

File: packages/splits-kit/src/components/CreateSplit/CreateSplitForm.tsx [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+  defaultRecipients: Recipient[]
+  defaultDistributorFeeOptions: number[]
+  onSuccess?: (events: Log[]) => void
+}) => {
+  const { createSplit, error } = useCreateSplit()
+  // NOTE: should we log this error?
+  if (error) {
+    console.error(error)
+  }
+
```
Comment (shrimpynuts) [inline packages/splits-kit/src/components/CreateSplit/CreateSplitForm.tsx:41]: Helps to debug but not sure this is the best way to propagate error messages

Comment (mihoward21) [reply packages/splits-kit/src/components/CreateSplit/CreateSplitForm.tsx:41]: should we have an `onError` option for create split too?

File: packages/splits-kit/src/constants/chains.ts [hunk 4]
```diff
@@ -0,0 +1,638 @@
...
+  readonly distributionThresholdRange?: [number, number, number]
+  readonly gasPriceGwei?: number
+}
+
+export type ChainInfo = {
+  readonly [chainId in SupportedChainId]: L1ChainInfo
+}
+
+// merge with @usedapp & own utils getExplorer etc
+export const CHAIN_INFO: ChainInfo = {
```
Comment (mihoward21) [inline packages/splits-kit/src/constants/chains.ts:126]: couple questions here:
- is this stuff used in the actual sdk? or just for storybook?
- and i think not all of the data is needed, right? like certainly the `supportsSwapper` field isn't needed. If this stuff is just used for storybook then less important. If it's being included in the sdk package though then I think trying to trim it down to what's necessary seems good.

File: packages/splits-sdk/src/utils/balances.ts [hunk 5]
```diff
@@ -111,7 +111,6 @@ const processBalanceMulticallResponse: (
     if (token === ADDRESS_ZERO) {
       balances[ADDRESS_ZERO] = balance
     } else {
-      if (!tokenData[token]) return // Unable to fetch token data
```
Comment (mihoward21) [inline packages/splits-sdk/src/utils/balances.ts:114]: this seems wrong. i assume you just need to pull in master or something?

File: packages/splits-sdk/src/client/base.ts [hunk 6]
```diff
@@ -466,8 +472,10 @@ export class BaseClientMixin extends BaseTransactions {
 }
 
 export class BaseGasEstimatesMixin extends BaseTransactions {
-  async multicall(multicallArgs: MulticallConfig): Promise<bigint> {
-    const gasEstimate = await this._multicallTransaction(multicallArgs)
+  async multicall({ calls }: { calls: CallData[] }): Promise<bigint> {
```
Comment (mihoward21) [inline packages/splits-sdk/src/client/base.ts:475]: seems like all these multicall changes that got merged in earlier today are getting reverted.

**Comment by mihoward21** [comment]
I think this all looks good. I think it looks like things are separated / set up to keep the storybook stuff out of the actual npm package.

Only major issues is it looks like it's reverting some recent changes. Maybe just need to pull in master?

File: packages/splits-kit/src/constants/chains.ts (hunk 4 above)
Comment (shrimpynuts) [reply packages/splits-kit/src/constants/chains.ts:126]: It's actually used in both, I will prune it down.

File: packages/splits-sdk/src/utils/balances.ts (hunk 5 above)
Comment (shrimpynuts) [reply packages/splits-sdk/src/utils/balances.ts:114]: Yeah I'll merge main in again

File: packages/splits-kit/src/components/CreateSplit/CreateSplitForm.tsx (hunk 3 above)
Comment (shrimpynuts) [reply packages/splits-kit/src/components/CreateSplit/CreateSplitForm.tsx:41]: Yeah good idea

**Pull request merged**